pin/gc/gc.go

```diff
@@ -104,6 +105,18 @@ func GC(ctx context.Context, bs bstore.GCBlockstore, ls dag.LinkService, pn pin.
 		if errors {
 			output <- Result{Error: ErrCannotDeleteSomeBlocks}
 		}
+
+		defer log.EventBegin(ctx, "GC.datastore").Done()
+		gds, ok := ds.(dstore.GCDatastore)
+		if !ok {
+			return
+		}
+
+		err = gds.CollectGarbage()
+		if err != nil {
+			output <- Result{Error: err}
```

Comment:
This can deadlock if the context has been canceled and output is full. However, we already *have* this bug (above) so feel free to fix it separately. See #4593.